sympy/parsing/mathematica.py

```diff
@@ -8,126 +9,218 @@ def mathematica(s):
     return sympify(parse(s))
 
 
-class _Parse(object):
-    '''An instance of this class converts basic Mathematica expression to
-    Python.'''
+class parse(object):
+    '''parse class converts basic Mathematica expression to Python.'''
 
-    replaces1 = (
-        (' ', ''),
-        ('^', '**'),
-    )
+    # translate
+    F = {
+        # (func in Mathematica, number of arguments): func in sympy
+        ('Mod', 2): 'Mod',
+        ('Sqrt', 1): 'sqrt',
+        ('Exp', 1): 'exp',
+        ('Log', 1): 'log',
+    }
 
-    replaces2 = (
-        ('[', '('),
-        (']', ')'),
-    )
+    # trigonometric, e.t.c.
+    for arc, tri, h in product(('', 'Arc'), (
+            'Sin', 'Cos', 'Tan', 'Cot', 'Sec', 'Csc'), ('', 'h')):
 
-    rules1 = (
-        # 'Arc' to 'a'
-        (r'''
-                (               # group1
-                .*              # any characters
-                [^a-zA-Z]       # a single character except alphabet
-                )
-                Arc             # detect 'Arc'
-                (               # group2
-                [a-zA-Z]        # a single alphabet
-                .*              # any characters
-                )
-                ''',
-         lambda m: m.group(1) + 'a' + m.group(2)),
-
-        # 'Arc' (at the top) to 'a'
-        (r'''
-                \AArc           # detect 'Arc' at the top of the string
-                (               # group1
-                [a-zA-Z]        # a single alphabet
-                .*              # any characters
-                )
-                ''',
-         lambda m: 'a' + m.group(1)),
+        fm = arc + tri + h
+
+        if arc:  # arc func
+            fs = 'a' + tri.lower() + h
+
+        else:    # non-arc func
+            fs = tri.lower() + h
+
+        F.update({(fm, 1): fs})
+
+    F_swap = {
+        # when you say log(2,4) in Mathematica, base is 2.
+        ('Log', 2): 'log',
```

Comment:
Would there be any advantage to introducing a `logb` wrapper function in SymPy with the syntax `logb(b, a) = log(a, b)`? Then Mathematic `Log(x, y)` could be passed directly to `logb(x, y)`

Reply:
It would be a nice idea if only this logarithmic function needs an argument swap. But, I have no idea how many functions there are which need such a treatment. I do not know if it is the best choice to introduce a wrapper function in SymPy every time when facing this kind of matters.

Reply:
I think that would just be confusing. Mathematica clearly chose its order because when you write log_b(x) the base is written before the argument. SymPy uses its convention because the base is an optional argument (defaulting to E), and in Python optional arguments go after required arguments. 

There are other SymPy functions where the second argument is optional, or where the argument order is effectively arbitrary, and I wouldn't be surprised if Mathematica chose a different convention for some of them as well.

Reply:
In my opinion, a parsing function should be written without affecting any other SymPy files.